### PR TITLE
GH-2032: As a smith I want to integrate the n4jsd-generator setup into StdLib/OPR

### LINKED ISSUE #2032: As a smith I want to integrate the n4jsd-generator setup into StdLib/OPR
This userstory is about integrating the n4jsd-generator into internal projects. While doing that, the n4jsd-generator will be improved here and there regarding some small features to make a first integration work.

### PR DESCRIPTION
**Pull request by mmews-n4**
closes #2032

### CODE REVIEW

**Comment by mmews-n4** [comment]
I integrated @types/rc and @types/deepEqual into StdLib and think that this user story can be seen as done now. I could not find any other npms to integrate into OPR or StdLib due to the reduced feature set of the n4jsd-generator. I think as a follow-up it would be a practical approach to chose a single or multiple npms to replace as a next step. At the moment, I suggest the following candidates:
* moment
* moment-timezone
* js.cookie
* geckodriver
* jsdom
* fs-extra
* debug
* async-throttle
This list was created manually by looking through their @n4jsd projects and what feature set they need.

File: n4js-tools/@n4js-temp/n4jsd-generator/src/n4js/dts2n4jsd.n4js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env node
+#!/usr/bin/env node -r esm
```
Comment (mor-n4) [inline n4js-tools/@n4js-temp/n4jsd-generator/src/n4js/dts2n4jsd.n4js:1]: Thanks!

File: n4js-libs/packages/n4js-cli/src/n4js/n4jsc.n4js [hunk 2]
```diff
@@ -12,8 +12,6 @@
  */
 "use strict";
 
+import {runN4jscSync} from "n4js-cli";
 
-require = require("esm")(module);
-let n4jscli = require("../src-gen/index.js");
-
-n4jscli.runN4jscSync();
+runN4jscSync();
```
Comment (mor-n4) [inline n4js-libs/packages/n4js-cli/src/n4js/n4jsc.n4js:17]: Nice to have this aligned!

**Review by mor-n4 — APPROVED** [review]
:+1:

**Comment by mmews-n4** [comment]
Thanks for the review! Builds green. Merging.

**Pull request merged**